### PR TITLE
TASK4:

### PR DESCRIPTION
**Pull request by AndreyKirshin**
added and confugured jest & enzyme
written tests on all components
added cypress
written one e2e test

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -61,3 +61,5 @@ typings/
 # next.js build output
 .next
 
+__snapshots__/
```
Comment (SKolevatov) [inline .gitignore]: You should commit snapshots, otherwise what's the profit of having them?

File: package.json [hunk 2]
```diff
@@ -31,17 +34,25 @@
...
+    "enzyme-adapter-react-16": "^1.12.0",
     "extract-text-webpack-plugin": "^4.0.0-beta.0",
     "html-webpack-plugin": "3.2.0",
+    "identity-obj-proxy": "^3.0.0",
+    "install": "^0.12.2",
     "jest": "^24.5.0",
     "json-loader": "^0.5.7",
     "less": "^3.9.0",
     "less-loader": "^4.1.0",
+    "npm": "^6.9.0",
```
Comment (SKolevatov) [inline package.json]: Why do you need it?

File: src/client/view/movieList/movieList.test.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import { shallow, mount } from 'enzyme';
+import MovieList from './movieList';
+import MovieItem from '../movieItem/movieItem';
+
+describe('MovieList component', () => {
+
+    let data;
+    let onClick;
+    beforeAll(() => {
+        data = [{
```
Comment (SKolevatov) [inline src/client/view/movieList/movieList.test.jsx:12]: I would suggest to create a file with movie mocks. In all your tests you may then just invoke a method like:
`getMockedMovies(numberOfMovies)`

Comment (SKolevatov) [reply src/client/view/movieList/movieList.test.jsx:12]: this is not crucial at all, btw

File: .gitignore (hunk 1 above)
Comment (AndreyKirshin) [reply .gitignore]: corrected

File: package.json (hunk 2 above)
Comment (AndreyKirshin) [reply package.json]: removed

**Pull request merged**